sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonPropertyNameSerializerBuilder.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.google.gson.GsonBuilder;
+
+/**
+ * Fluent builder class that configures and instantiates instances of {@link GsonPropertyNameSerializer}.
+ */
+public class GsonPropertyNameSerializerBuilder {
```

Comment:
If we have `GsonJsonSerializer` implement the `PropertyNameSerializer` interface we won't need this class, which feels very duplicative of `GsonJsonSerializerBuilder`.

Reply:
Do you mean we reuse GsonJsonSerializerBuilder to build `PropertyNameSerializer`?
We need the Gson instance in `PropertyNameSerializer` to check the excluder

Reply:
We have removed the independent PropertyNameSerializer impl from Gson/Jackson